helix-view/src/clipboard.rs

```diff
@@ -243,12 +242,37 @@ pub mod provider {
 
         #[cfg(not(any(windows, target_os = "macos")))]
         pub fn is_exit_success(program: &str, args: &[&str]) -> bool {
-            std::process::Command::new(program)
-                .args(args)
-                .output()
-                .ok()
-                .and_then(|out| out.status.success().then(|| ())) // TODO: use then_some when stabilized
-                .is_some()
+            use helix_lsp::block_on;
+            use tokio::process::Command;
+            use tokio::time::timeout;
+
+            async fn is_exit_success_async(program: &str, args: &[&str]) -> bool {
+                Command::new(program)
+                    .args(args)
+                    .kill_on_drop(true)
+                    .output()
+                    .await
+                    .ok()
+                    .and_then(|out| out.status.success().then(|| ()))
+                    .is_some()
+            }
+
+            let task = async {
+                match timeout(
+                    std::time::Duration::from_secs(1),
+                    is_exit_success_async(program, args),
+                )
+                .await
+                {
+                    Ok(success) => success,
+                    Err(_) => {
+                        log::info!("Timed out waiting for {}", program);
+                        false
+                    }
+                }
+            };
+
+            block_on(task)
```

Comment:
Needs to be wrapped with https://docs.rs/tokio/latest/tokio/task/fn.block_in_place.html